contracts/StakeManager.sol

```diff
@@ -222,9 +222,18 @@ contract StakeManager is Ownable {
         if (deltaTime < MIN_LOCKUP_PERIOD || deltaTime > MAX_LOCKUP_PERIOD) {
             revert StakeManager__InvalidLockTime();
         }
-        _mintInitialMP(account, _timeToIncrease, 0);
-        //update account storage
+
+        uint256 mpToMint = _getMaxMPToMint(
+            _getMPToMint(account.balance, _timeToIncrease),
+            account.balance,
+            account.initialMP,
+            account.currentMP
+        );
```

Comment:
Instead we want to mint whatever MPs the account is entitled for.

@3esmit this ultimately relies on `getMPToMin(account.balance, _timeToIncrease)` just like before.
However, I'm wondering if we shouldn't instead do

```
_getMPToMint(account.balance, _timeToIncrease - account.lastMint)
```
What do you think?